src/Nancy/Responses/DefaultJsonSerializer.cs

```diff
@@ -100,6 +100,7 @@ private static bool IsJsonType(string contentType)
             return contentMimeType.Equals("application/json", StringComparison.OrdinalIgnoreCase) ||
                 contentMimeType.StartsWith("application/json-", StringComparison.OrdinalIgnoreCase) ||
                 contentMimeType.Equals("text/json", StringComparison.OrdinalIgnoreCase) ||
+                contentMimeType.Equals("application/problem+json", StringComparison.OrdinalIgnoreCase) ||
                 (contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase) &&

```

Comment:
Yeah, it's not a serializers job to validate your media types. It should only care about the structured syntax suffix.